src/main/java/org/eclipse/pathfinder/api/GraphTraversalService.java

```diff
@@ -50,53 +50,34 @@ public List<TransitPath> findShortestPath(
       @Size(min = 8, max = 8, message = "Deadline value must be eight characters long.")
           @QueryParam("deadline")
           String deadline) {
-    LocalDateTime date = nextDate(LocalDateTime.now());
 
     List<String> allVertices = dao.listLocations();
     allVertices.remove(originUnLocode);
     allVertices.remove(destinationUnLocode);
 
     int candidateCount = getRandomNumberOfCandidates();
-    List<TransitPath> candidates = new ArrayList<>(candidateCount);
+    List<TransitPath> candidates = new ArrayList<>();
 
     for (int i = 0; i < candidateCount; i++) {
       allVertices = getRandomChunkOfLocations(allVertices);
-      List<TransitEdge> transitEdges = new ArrayList<>(allVertices.size() - 1);
-      String firstLegTo = allVertices.get(0);
-
-      LocalDateTime fromDate = nextDate(date);
-      LocalDateTime toDate = nextDate(fromDate);
-      date = nextDate(toDate);
-
-      transitEdges.add(
-          new TransitEdge(
-              dao.getVoyageNumber(originUnLocode, firstLegTo),
-              originUnLocode,
-              firstLegTo,
-              fromDate,
-              toDate));
-
-      for (int j = 0; j < allVertices.size() - 1; j++) {
-        String current = allVertices.get(j);
-        String next = allVertices.get(j + 1);
-        fromDate = nextDate(date);
-        toDate = nextDate(fromDate);
-        date = nextDate(toDate);
+      List<TransitEdge> transitEdges = new ArrayList<>();
```

Comment:
Please revert this. It is inefficient to create an empty list when capacity is known.